File: src/main/java/no/nav/veilarbaktivitet/stilling_fra_nav/DelingAvCvManueltAvbruttService.java
package no.nav.veilarbaktivitet.stilling_fra_nav;

import io.micrometer.core.annotation.Timed;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import no.nav.veilarbaktivitet.aktivitet.domain.AktivitetData;
import no.nav.veilarbaktivitet.person.Person;
import org.springframework.stereotype.Service;

import java.util.List;

@Slf4j
@Service
@RequiredArgsConstructor
public class DelingAvCvManueltAvbruttService {

    private final DelingAvCvService delingAvCvService;
    private final DelingAvCvDAO delingAvCvDAO;

    @Timed(value = "stillingFraNavAvbruttEllerFullfortUtenSvar", histogram = true)
    public int notifiserFullfortEllerAvbruttUtenSvar(int maxantall) {
        List<AktivitetData> aktivitetData = delingAvCvDAO.hentStillingFraNavSomErFullfortEllerAvbruttUtenSvar(maxantall);
        aktivitetData.forEach(aktivitet -> {
            try {
                delingAvCvService.notifiserAvbruttEllerFullfortUtenSvar(aktivitet, Person.navIdent("SYSTEM"));
            } catch (Exception e) {
                log.warn("Behandling av fullført/avbrutt aktivitet aktivitetId={} feilet", aktivitet.getId());
                log.error("Kunne ikke behandle avbrutt/fullført aktivitet", e);
            }
        });
        return aktivitetData.size();
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/stilling_fra_nav/DelingAvCvManueltAvbruttService.kt
package no.nav.veilarbaktivitet.stilling_fra_nav

import io.micrometer.core.annotation.Timed
import lombok.RequiredArgsConstructor
import no.nav.veilarbaktivitet.aktivitet.domain.AktivitetData
import no.nav.veilarbaktivitet.util.BatchJob
import no.nav.veilarbaktivitet.util.BatchResult
import no.nav.veilarbaktivitet.util.BatchTrackingDAO
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
open class DelingAvCvManueltAvbruttService(
    private val delingAvCvService: DelingAvCvService,
    private val delingAvCvDAO: DelingAvCvDAO,
    private val batchTrackingDao: BatchTrackingDAO
) {
    private val log = LoggerFactory.getLogger(javaClass)

    @Timed(value = "stillingFraNavAvbruttEllerFullfortUtenSvar", histogram = true)
    open fun notifiserFullfortEllerAvbruttUtenSvar(maxantall: Int): Int {
        return batchTrackingDao.withOffset(BatchJob.Deling_av_cv_avbrutt_eller_fuulfort_uten_svar) { sisteProsesserteVersjonFørBatch ->
            val aktivitetData = delingAvCvDAO.hentStillingFraNavSomErFullfortEllerAvbruttUtenSvar(maxantall.toLong(), sisteProsesserteVersjonFørBatch)
            aktivitetData.map { aktivitet: AktivitetData ->
                try {
                    delingAvCvService.notifiserAvbruttEllerFullfortUtenSvar(aktivitet)
                    return@map BatchResult.Success(aktivitet.versjon)
                } catch (e: Exception) {
                    log.warn("Behandling av fullført/avbrutt aktivitet aktivitetId=${aktivitet.id} feilet")
                    log.error("Kunne ikke behandle avbrutt/fullført aktivitet", e)
                    return@map BatchResult.Failure(aktivitet.versjon)
                }
            }
        }.size.also { log.info("Avsluttet $it brukernotifikasjoner for stilling-fra-nav aktiviteter som ble manuelt avbrutt/fullført") }
    }
}
